tests/tensor/test_basic.py

```diff
@@ -3598,52 +3591,66 @@ def test_diag(self):
             # The right matrix is created
             assert (r == v).all()
 
-        # Test scalar input
-        xx = scalar()
-        with pytest.raises(ValueError):
-            diag(xx)
-
         # Test passing a list
         xx = [[1, 2], [3, 4]]
         g = diag(xx)
         f = function([], g)
         assert np.array_equal(f(), np.diag(xx))
 
-    def test_infer_shape(self):
-        rng = np.random.default_rng(utt.fetch_seed())
+    @pytest.mark.parametrize("inp", (scalar, tensor3))
+    def test_diag_invalid_input_ndim(self, inp):
+        x = inp()
+        with pytest.raises(ValueError, match="Input must be 1- or 2-d."):
+            diag(x)
 
-        x = vector()
-        g = diag(x)
-        f = pytensor.function([x], g.shape)
-        topo = f.maker.fgraph.toposort()
-        if config.mode != "FAST_COMPILE":
-            assert sum(isinstance(node.op, AllocDiag) for node in topo) == 0
-        for shp in [5, 0, 1]:
-            m = rng.random(shp).astype(self.floatX)
-            assert (f(m) == np.diag(m).shape).all()
 
-        x = matrix()
-        g = diag(x)
+class TestExtractDiag:
+    @pytest.mark.parametrize("axis1, axis2", [(0, 1), (1, 0)])
+    @pytest.mark.parametrize("offset", (-1, 0, 2))
+    def test_infer_shape(self, offset, axis1, axis2):
+        rng = np.random.default_rng(utt.fetch_seed())
+
+        x = matrix("x")
+        g = ExtractDiag(offset=offset, axis1=axis1, axis2=axis2)(x)
         f = pytensor.function([x], g.shape)
         topo = f.maker.fgraph.toposort()
         if config.mode != "FAST_COMPILE":
             assert sum(isinstance(node.op, ExtractDiag) for node in topo) == 0
         for shp in [(5, 3), (3, 5), (5, 1), (1, 5), (5, 0), (0, 5), (1, 0), (0, 1)]:
-            m = rng.random(shp).astype(self.floatX)
-            assert (f(m) == np.diag(m).shape).all()
+            m = rng.random(shp).astype(config.floatX)
+            assert (
+                f(m) == np.diagonal(m, offset=offset, axis1=axis1, axis2=axis2).shape
+            ).all()
 
-    def test_diag_grad(self):
+    @pytest.mark.parametrize("axis1, axis2", [(0, 1), (1, 0)])
+    @pytest.mark.parametrize("offset", (0, 1, -1))
+    def test_grad_2d(self, offset, axis1, axis2):
+        diag_fn = ExtractDiag(offset=offset, axis1=axis1, axis2=axis2)
         rng = np.random.default_rng(utt.fetch_seed())
-        x = rng.random(5)
-        utt.verify_grad(diag, [x], rng=rng)
         x = rng.random((5, 3))
-        utt.verify_grad(diag, [x], rng=rng)
+        utt.verify_grad(diag_fn, [x], rng=rng)
+
+    @pytest.mark.parametrize(
+        "axis1, axis2",
+        [
+            (0, 1),
+            (1, 0),
+            (1, 2),
+            (2, 1),
+            (0, 2),
+            (2, 0),
```

Comment:
`(0, 2)` and `(2, 0)` are the cases that would previously have raised, correct?

Reply:
Before this PR anything above 2d input would have raised. The cases you listed weren't supported in the previous iteration of this PR